src/cli/commands/upgrade.js

```diff
@@ -19,6 +20,20 @@ export async function run(
   args: Array<string>,
 ): Promise<void> {
   const lockfile = args.length ? await Lockfile.fromDirectory(config.cwd, reporter) : new Lockfile();
-  const install = new Add(args, flags, config, reporter, lockfile);
+  let addArgs = [...args];
+
+  if (addArgs.length) {
+    const [dependency] = addArgs;
```

Comment:
This would not work with that.

I actually was curious whether that was a use case.

And also whether `args` / `addArgs` could be composed of other types of arguments (e.g. flags). 

If I assumed `args` consisted of only packages, I can easily switch this to a map-like function.